### PR TITLE
[DOCS-2772] Add APISIX-Datadog doc and pics

### PR DESCRIPTION
**Pull request by yzeng25**
### What does this PR do?

Add document for Apache APISIX-Datadog plugin.
### Motivation

 APISIX-Datadog plugin was released a couple days ago, thought it would be good to post and let others know about it.
### Preview

https://docs-staging.datadoghq.com/<BRANCH_NAME>/<PATH>

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by yzeng25** [comment]
I tried to upload the markdown file but it kept getting ignored. Does anyone has ideas about this failure?

**Comment by moonming** [comment]
> I tried to upload the markdown file but it kept getting ignored. Does anyone has ideas about this failure?

please take a look at this file: https://github.com/DataDog/documentation/blob/master/.gitignore#L60

**Comment by yzeng25** [comment]
> > I tried to upload the markdown file but it kept getting ignored. Does anyone has ideas about this failure?
> 
> please take a look at this file: https://github.com/DataDog/documentation/blob/master/.gitignore#L60

Cool. I added a line `!content/en/integrations/apisix.md` and it works now, thanks :)

**Comment by ruthnaebeck** [comment]
@yzeng25 - Thanks for the PR! We will look into this.

**Comment by apigirl** [comment]
Hi @yzeng25 - thank you for this PR! Where has this plugin been released? I do not see it here: http://apisix.apache.org/plugins

File: content/en/integrations/apisix.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+If you are using Kubernetes in your production environment, you can deploy `dogstatsd` as a `Daemonset` or as a `Multi-Container Pod` alongside Apache APISIX agent.
+
+## Activate APISIX-Datadog plugin
+
+The following is an example on how to activate the datadog plugin for a specific route. We are assumming your `dogstatsd` agent is already up an running.
+
+```shell
+# enable plugin for a specific route
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```
Comment (apigirl) [inline content/en/integrations/apisix.md:65]: hi @yzeng25 - can you please confirm if this is a real API key? If so, please delete the API key as soon as possible. I am also initiating an internal process to investigate as well.

Comment (moonming) [reply content/en/integrations/apisix.md:65]: hi @apigirl it's not a real key, just a mock key for APISIX, and it's nothing with Datadog.
@yzeng25 please add docs and comments for this.

**Comment by moonming** [comment]
> Hi @yzeng25 - thank you for this PR! Where has this plugin been released? I do not see it here: http://apisix.apache.org/plugins

Datadog plugin merged into GitHub repo: https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/datadog.md, and we will update this plugin in the website too. Thanks for your reminder.

File: content/en/integrations/apisix.md (hunk 1 above)
Comment (yzeng25) [reply content/en/integrations/apisix.md:65]: Sure @moonming

**Comment by yzeng25** [comment]
> Hi @yzeng25 - thank you for this PR! Where has this plugin been released? I do not see it here: http://apisix.apache.org/plugins

Hi @apigirl ,the plugin is here: http://apisix.apache.org/docs/apisix/next/plugins/datadog

**Comment by apigirl** [comment]
great, thank you for the information!

File: content/en/integrations/apisix.md (hunk 1 above)
Comment (apigirl) [reply content/en/integrations/apisix.md:67]: okay! thank you for verifying

**Comment by yzeng25** [comment]
Hi @apigirl , we have posted this blog on our website. Would you kindly reference to our website if possible? Thanks!

ref: https://apisix.apache.org/blog/2021/11/12/apisix-datadog

**Comment by apigirl** [comment]
hi @yzeng25 - we've added a separate PR here to update this in the way that @jeremy-lq indicated in his email response to your original request.

**Pull request closed without merging**